kasa/smart/modules/lightstripeffect.py

```diff
@@ -0,0 +1,109 @@
+"""Module for light effects."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from ...interfaces.lighteffect import LightEffect as LightEffectInterface
+from ..effects import EFFECT_MAPPING, EFFECT_NAMES
+from ..smartmodule import SmartModule
+
+if TYPE_CHECKING:
+    pass
+
+
+class LightStripEffect(SmartModule, LightEffectInterface):
+    """Implementation of dynamic light effects."""
+
+    REQUIRED_COMPONENT = "light_strip_lighting_effect"
+
+    @property
+    def name(self) -> str:
+        """Name of the module."""
+        return "LightEffect"
```

Comment:
Hmm, is this really necessary?

Reply:
Yes because the effects are accessed in the dictionary by the key which is the module name.  We have the `LightStripEffect` and `LightEffect` implementing the same common interface on the assumption that a device will either support dynamic light effects or light strip effects but not both (or if they ever support both we give one preference over the other)

Reply:
I'd rather avoid such magic, if possible, as it can lead to hard to debug behaviors. But if this is the way to go, please add a comment to describe why this was done to help future ourselves :-)

Reply:
Comment added